raiden/transfer/channel.py

```diff
@@ -388,6 +392,172 @@ def is_valid_lockedtransfer(
+
+        elif received_balance_proof.locksroot != locksroot_without_lock:
+            # The locksroot must be updated to include the new lock
+            msg = (
+                'Invalid {} message. '
+                "Balance proof's locksroot didn't match, expected: {} got: {}."
+            ).format(
+                message_name,
+                hexlify(locksroot_without_lock).decode(),
+                hexlify(received_balance_proof.locksroot).decode(),
+            )
+
+            result = (False, msg, None)
+
+        elif received_balance_proof.transferred_amount != current_transferred_amount:
+            # Mediated transfers must not change transferred_amount
+            msg = (
+                'Invalid {} message. '
+                "Balance proof's transferred_amount changed, expected: {} got: {}."
+            ).format(
+                message_name,
+                current_transferred_amount,
+                received_balance_proof.transferred_amount,
+            )
+
+            result = (False, msg, None)
+
+        elif not is_valid_amount(sender_state, lock.amount):
+            # We can validate that the Unlock message will have a transferred
+            # amount, accepting a lock that cannot be unlocked is useless
+            msg = (
+                "Invalid {} message. "
+                "Unlocking the lock would result in an overflow. max: {} result would be: {}"
+            ).format(
+                message_name,
+                UINT256_MAX,
+                transferred_amount_after_unlock,
+            )
+
+            result = (False, msg, None)
+
+        elif received_balance_proof.locked_amount != expected_locked_amount:
+            # Mediated transfers must increase the locked_amount by lock.amount
+            msg = (
+                'Invalid {} message. '
+                "Balance proof's locked_amount is invalid, expected: {} got: {}."
+            ).format(
+                message_name,
+                expected_locked_amount,
+                received_balance_proof.locked_amount,
+            )
+
+            result = (False, msg, None)
+
+        elif received_balance_proof.channel_identifier != channel_state.identifier:
+            # The balance proof must be tied to this channel, otherwise the
+            # on-chain contract would be sucesstible to replay attacks across
+            # channels.
+            msg = (
+                'Invalid {} message. '
+                'Balance proof is tied to the wrong channel, expected: {} got: {}'
+            ).format(
+                message_name,
+                channel_state.identifier,
+                received_balance_proof.channel_identifier,
+            )
+            result = (False, msg, None)
+
+        # the locked amount is limited to the current available balance, otherwise
+        # the sender is attempting to game the protocol and do a double spend
+        elif lock.amount > distributable:
```

Comment:
This check is really wrong, in it's current form it's not allowed to remove a lock if the sender doesn't have capacity.

Reply:
This would be removed. I figure that as the locked amount was at some point a balance that was locked by a certain transfer that now expired.